pkg/certificate/secret.go

```diff
@@ -42,19 +42,18 @@ func populateTLSSecret(secret corev1.Secret, keyPair *triple.KeyPair) *corev1.Se
 		corev1.TLSCertKey:       triple.EncodeCertPEM(keyPair.Cert),
 		corev1.TLSPrivateKeyKey: triple.EncodePrivateKeyPEM(keyPair.Key),
 	}
-	secret.Type = corev1.SecretTypeTLS
 	return &secret
 }
 
 func (m *Manager) applyTLSSecret(secret types.NamespacedName, keyPair *triple.KeyPair) error {
-	return m.applySecret(secret, keyPair, populateTLSSecret)
+	return m.applySecret(secret, corev1.SecretTypeTLS, keyPair, populateTLSSecret)
 }
 
 func (m *Manager) applyCASecret(secret types.NamespacedName, keyPair *triple.KeyPair) error {
-	return m.applySecret(secret, keyPair, populateCASecret)
+	return m.applySecret(secret, corev1.SecretTypeOpaque, keyPair, populateCASecret)
```

Comment:
isn't it strange that there is no `corev1.SecretTypeCA`?

Reply:
It's not, I though the same.